File: LeetCode/2114_maximum_number_of_words_found_in_sentences.py
#!/usr/bin/env python3
# pylint: disable=invalid-name, too-few-public-methods, unused-import, no-self-use,missing-function-docstring,consider-using-enumerate
'''
    LeetCode solution class
'''
from typing import List, Optional


class Solution:
    '''
    LeetCode solution class
    '''

    def mostWordsFound(self, sentences: List[str]) -> int:
        max_words = 0
        for sentence in sentences:
            words = len(sentence.split(" "))
            max_words = words if words > max_words else max_words
        return max_words


solution = Solution()
print(solution.mostWordsFound(["alice and bob love leetcode",
      "i think so too", "this is great thanks very much"]))  # 6
print(solution.mostWordsFound(
    ["please wait", "continue to fight", "continue to win"]))  # 3

File: LeetCode/2114_maximum_number_of_words_found_in_sentences.php
<?php

declare(strict_types=1);

use Solution as GlobalSolution;

/**
 * Template Soultion class for LeetCode problems
 * 
 */
class Solution
{
    /**
     * @param array $sentences
     * @return int
     */
    public function mostWordsFound(array $sentences): int
    {
        $max = 0;
        foreach ($sentences as $sentence) {
            $words = count(explode(" ", $sentence));
            $max = ($words > $max) ? $words : $max;
        }
        return $max;
    }
}
// test suite
$solution = new GlobalSolution();

echo ($solution->mostWordsFound([
    "alice and bob love leetcode",
    "i think so too",
    "this is great thanks very much"
])); // 6
